tensorflow_probability/python/experimental/mcmc/sample_fold_test.py

```diff
@@ -0,0 +1,301 @@
+# Copyright 2020 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""Tests for drivers in a Streaming Reductions Framework."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+# Dependency imports
+from absl.testing import parameterized
+import collections
+import numpy as np
+
+import tensorflow.compat.v2 as tf
+import tensorflow_probability as tfp
+from tensorflow_probability.python.internal import samplers
+from tensorflow_probability.python.internal import test_util
+
+
+TestTransitionKernelResults = collections.namedtuple(
+    'TestTransitionKernelResults', 'counter_1, counter_2')
+
+
+class TestTransitionKernel(tfp.mcmc.TransitionKernel):
+  "Deterministic Transition Kernel for testing purposes"
+
+  def __init__(self, shape=(), is_calibrated=True, accepts_seed=True):
+    self._shape = shape
+    self._is_calibrated = is_calibrated
+    self._accepts_seed = accepts_seed
+
+  def one_step(self, current_state, previous_kernel_results, seed=None):
+    if seed is not None and not self._accepts_seed:
+      raise TypeError('seed arg not accepted')
+    return current_state + tf.ones((self._shape)), TestTransitionKernelResults(
+        counter_1=previous_kernel_results.counter_1 + tf.ones(self._shape),
+        counter_2=previous_kernel_results.counter_2 + tf.ones(self._shape) * 2)
+
+  def bootstrap_results(self, current_state):
+    return TestTransitionKernelResults(
+        counter_1=tf.zeros(self._shape),
+        counter_2=tf.zeros(self._shape))
+
+  @property
+  def is_calibrated(self):
+    return self._is_calibrated
+
+
+class RandomTransitionKernel(tfp.mcmc.TransitionKernel):
```

Comment:
Do you think it's time to make a separate module collecting these identical, replicated test `TransitionKernel`s?  It's a tradeoff between explicitness and code de-duplication.

Reply:
Yeah, I've been thinking the same thought more and more as I've been using these objects more and more. Would grouping them with other test_utils in a separate PR be appropriate?

Reply:
I'm ambivalent about whether to call them test_utils or test_fixtures, since they are sort of the latter rather than the former.  Separate PR is reasonable.